niceman/distributions/conda.py

```diff
@@ -92,13 +92,43 @@ def install_packages(self, session=None):
             Environment sub-class instance.
         """
 
-        # TODO: Need to figure out a graceful way to install conda before we can install packages here.
-        # for package in self.provenance['packages']:
-        #     session.add_command(['conda',
-        #                            'install',
-        #                            package['name']])
+        # TODO: Install Conda
+#        print("We got" + str(environment))
+        # Loop through non-root packages, creating the conda-env export
+        # file to import
+        for env in self.environments:
+            if env.path == self.path:
+                continue
+            export_contents = self.create_conda_export(env)
+            print ("EXPORT\n" + export_contents)
+
         return
 
+    @staticmethod
+    def create_conda_export(env):
+        # TODO: The environment name should be discovered on retrace
+        name = os.path.basename(os.path.normpath(env.path))
+        export = "name: %s\n" % name
+        # Collect channels
+        export += "channels:\n"
+        for c in env.channels:
+            export += "- %s\n" % c["name"]
+        export += "dependencies:\n"
+        # Collect packages (separating pip installed packages)
+        pip_export = ""
+        for p in env.packages:
+            if p.get("installer") is None:
+                export += "- %s=%s=%s\n" % (p["name"], p["version"],
+                                            p["build"])
+            if p.get("installer") == "pip":
```

Comment:
The only "installer" that's present in the environments is "pip",
correct?  Should we add an `else` branch for the `p["installer"] != "pip"`
case that logs a warning if another key is encountered?

So something like

```python
if "installer" not in p:
    ...
elif p["installer"] == "pip":
    ...
else:
    ...
```